tensorflow/core/ops/string_ops.cc

```diff
@@ -247,6 +247,57 @@ shape: a length-2 vector of int64 representing the shape of the sparse
   of tokens in a single input entry.
 )doc");
 
+REGISTER_OP("StringSplitUTF8")
+    .Input("input: string")
+    .Input("delimiter: string")
+    .Output("indices: int64")
+    .Output("values: string")
+    .Output("shape: int64")
+    .Attr("skip_empty: bool = true")
+    .SetShapeFn([](InferenceContext* c) {
+      ShapeHandle unused;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 1, &unused));
+      TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(1), 1, &unused));
+
+      c->set_output(0, c->Matrix(InferenceContext::kUnknownDim, 2));
+      c->set_output(1, c->Vector(InferenceContext::kUnknownDim));
+      c->set_output(2, c->Vector(2));
+      return Status::OK();
+    })
+    .Doc(R"doc(
+Split elements of `input` based on `delimiter` into a `SparseTensor`.
+
+Let N be the size of source (typically N will be the batch size). Split each
+element of `input` based on `delimiter` and return a `SparseTensor`
+containing the splitted tokens. Empty tokens are ignored.
+
+`delimiter` can be empty, or a string of split characters. If `delimiter` is an
+ empty string, each element of `input` is split into individual single-UTF-8
+ character strings, including splitting of UTF-8 multibyte sequences. Otherwise
+ every character of `delimiter` is a potential split point.
```

Comment:
Sorry, I should have caught this earlier - the documentation here says that `delimiter` can be a sequence of characters, in which case, it splits on any of them. However, the implementation does not seem to do that.

For the original `StringSplit` op:
- [this test](https://github.com/tensorflow/tensorflow/blob/r1.4/tensorflow/core/lib/strings/str_util_test.cc#L242), and
- [this test](https://github.com/tensorflow/tensorflow/blob/r1.4/tensorflow/python/kernel_tests/string_split_op_test.py#L110)
exercise the case where there are multiple delimiter characters specified.

So, long story short, unless I'm missing something - for `StringSplitUTF8`, delimiter cannot be a string of split characters, it has to be a single split character.

Is that correct? Instead of adding the complexity of handling multiple split characters for now (unless you're up for it), I'd change this to expect a single split character, and if the feature ever comes up then we'd allow for multiple.

Sound reasonable?

Reply:
Thanks @asimshankar. Given the original scope of #11399 I think that might be the best option for now. We can always add additional processing later if there is a request. The doc string has been updated.

Thanks for pointing this out and really appreciate that.